pallets/creditcoin/src/ocw/tasks.rs

```diff
@@ -0,0 +1,119 @@
+pub mod collect_coins;
+pub mod verify_transfer;
+
+use crate::ocw::errors::{OffchainError, VerificationResult};
+use crate::types::{
+	CollectedCoins, Task, TaskOutput, Transfer, UnverifiedCollectedCoins, UnverifiedTransfer,
+};
+use crate::{CollectedCoinsId, Config, TaskData, TransferId};
+use sp_runtime::traits::{UniqueSaturatedFrom, UniqueSaturatedInto};
+
+impl<AccountId, BlockNum, Hash, Moment> UnverifiedTransfer<AccountId, BlockNum, Hash, Moment>
+where
+	Moment: UniqueSaturatedInto<u64> + UniqueSaturatedFrom<u64>,
+	BlockNum: UniqueSaturatedInto<u64>,
+{
+	pub fn verify_ocw<T>(&self) -> VerificationResult<Option<T::Moment>>
+	where
+		T: Config<AccountId = AccountId, BlockNumber = BlockNum, Hash = Hash, Moment = Moment>,
+	{
+		crate::Pallet::<T>::verify_transfer_ocw(self)
+	}
+
+	pub fn into_output<T: Config>(
+		self,
+		timestamp: Option<T::Moment>,
+	) -> Transfer<AccountId, BlockNum, Hash, Moment>
+	where
+		T: Config<AccountId = AccountId, BlockNumber = BlockNum, Hash = Hash, Moment = Moment>,
+	{
+		Transfer { timestamp, ..self.transfer }
+	}
+
+	pub fn to_id<T: Config>(&self) -> TransferId<Hash>
```

Comment:
What's the upside?

Reply:
Oh I think I see, you mean why not implement `Into<TransferId>` instead of having `to_id`? 
Two reasons:

1. `Into` is consuming, so you'd have to `impl<'a> Into<TransferId> for &'a UnverifiedTransfer` to make it not take ownership of the whole thing.
2. You need an extra generic param (`T`) on the function plus some where bounds, and you can't add those to trait implementations if they're not in the trait definition.